packages/vite/src/node/config.ts

```diff
@@ -66,7 +67,13 @@ export function defineConfig(config: UserConfigExport): UserConfigExport {
   return config
 }
 
-export type PluginOption = Plugin | false | null | undefined | PluginOption[]
+export type PluginOption =
+  | Plugin
+  | false
+  | null
+  | undefined
+  | Promise<Plugin | false | null | undefined>
```

Comment:
after `asyncFlatten` the list would be flat, I think might be missing `PluginOption[]` 😁

Reply:
You're right again. This recursive thing is all too confusing 😄